components/tools/OmeroJava/test/integration/gateway/TablesFacilityTest.java

```diff
@@ -88,12 +96,92 @@ public void testGetSubsetTable() throws Exception {
+                if (type.equals(String.class)) {
+                    column[r] = "" + rand.nextInt();
+                } else if (type.equals(Long.class)) {
+                    column[r] = rand.nextLong();
+                } else if (type.equals(Double.class)) {
+                    column[r] = rand.nextDouble();
+                } else if (type.equals(Double[].class)) {
+                    Double[] d = new Double[4];
+                    for (int i = 0; i < 4; i++)
+                        d[i] = rand.nextDouble();
+                    column[r] = d;
+                }
+            }
+            data[c] = column;
+        }
+        
+        TableData td = new TableData(header, data);
+        long start = System.currentTimeMillis();
+        td = tablesFacility.addTable(rootCtx, ds, "Big Table", td);
+        long duration = System.currentTimeMillis() - start;
+        
+        System.out.println("Storing table with "+(cols*rows)+" entries took "+duration+" ms");
+        
+        duration = 0;
+        int nEntries = 0;
+        for(int i=0; i<10; i++) {
+            // read some random subsets from the table and compare to original data
+            int rowFrom = rand.nextInt(rows-1000); // can't request more than 1000 rows at once
+            int rowTo = rowFrom + rand.nextInt(1000);
+            int[] columns = new int[rand.nextInt(cols)];
+            
+            nEntries += (rowTo - rowFrom) * columns.length;
+            
+            for(int x=0; x<columns.length; x++)
+                columns[x] = rand.nextInt(cols);
+            
+            start = System.currentTimeMillis();
+            TableData td2 = tablesFacility.getTable(rootCtx, td.getOriginalFileId(),
+                    rowFrom, rowTo, columns);
+            duration += System.currentTimeMillis() - start;
+            
+            Object[][] data2 = td2.getData();
+            
+            for(int r=rowFrom; r<rowTo; r++) {
+                for(int c=0; c<columns.length; c++) {
+                    int index = columns[c];
+                    Class<?> type = header[index].getType();
+                    if (type.equals(String.class)) {
+                        Assert.assertEquals((String)data2[c][r-(int)td2.getOffset()], (String)data[index][r]);
+                    } else if (type.equals(Long.class)) {
+                        Assert.assertEquals((Long)data2[c][r-(int)td2.getOffset()], (Long)data[index][r]);
+                    } else if (type.equals(Double.class)) {
+                        Assert.assertEquals((Double)data2[c][r-(int)td2.getOffset()], (Double)data[index][r]);
+                    } else if (type.equals(Double[].class)) {
+                        Assert.assertEquals((Double[])data2[c][r-(int)td2.getOffset()], (Double[])data[index][r]);
+                    }
+                }
+            }
+        }
+        System.out.println("Requesting random subsets (average size: "+(nEntries/10)+" entries) took "+(duration/10)+" ms on average");
```

Comment:
you could add option like
``@Test(invocationCount = 5, threadPoolSize = 5)``
if you want to run the test few times, I imagine that the test will be over the timeout limit so you will probably have to add ``timeOut=Value``